frontend/src/pages/connectionTypes/ConnectionTypesTableRow.tsx

```diff
@@ -0,0 +1,119 @@
+  const createdDate = obj.metadata.creationTimestamp
+    ? new Date(obj.metadata.creationTimestamp)
+    : undefined;
+
+  const onUpdateEnabled = async (enabled: boolean) => {
+    setStatusMessage(enabled ? 'Enabling...' : 'Disabling...');
+    setErrorMessage(undefined);
+    pendingEnabledState.current = enabled ? 'true' : 'false';
+
+    const response = await updateConnectionTypeEnabled(obj, enabled);
+    if (response.success) {
+      onUpdate();
+      return;
+    }
+
+    setStatusMessage('Failed');
+    setErrorMessage(response.error || `Failed to ${enabled ? 'enable' : 'disable'}`);
+  };
+
+  React.useEffect(() => {
+    if (
+      pendingEnabledState.current !== undefined &&
+      obj.metadata.annotations?.['opendatahub.io/enabled'] === pendingEnabledState.current
+    ) {
+      setStatusMessage(undefined);
+      pendingEnabledState.current = undefined;
+    }
+  }, [obj.metadata.annotations]);
+
+  return (
+    <Tr>
+      <Td dataLabel="Name" width={70}>
+        <Flex direction={{ default: 'column' }} gap={{ default: 'gapSm' }}>
+          <TextContent>
+            <Text data-testid="connection-type-name">
+              {obj.metadata.annotations?.['openshift.io/display-name'] || obj.metadata.name}
+            </Text>
+          </TextContent>
+          <TextContent>
+            <Text
+              data-testid="connection-type-description"
+              style={{ color: 'var(--pf-v5-c-content--blockquote--Color)' }}
+            >
+              {obj.metadata.annotations?.['openshift.io/description']}
+            </Text>
+          </TextContent>
+        </Flex>
+      </Td>
+      <Td dataLabel="Creator" data-testid="connection-type-creator">
+        {obj.metadata.annotations?.['opendatahub.io/username'] === 'Pre-installed' ? (
+          <Label data-testid="connection-type-user-label">Pre-installed</Label>
+        ) : (
+          <TextContent>
+            <Text>{obj.metadata.annotations?.['opendatahub.io/username'] || 'unknown'}</Text>
+          </TextContent>
+        )}
+      </Td>
+      <Td dataLabel="Created" data-testid="connection-type-created">
+        <span style={{ whiteSpace: 'nowrap' }}>
+          <Timestamp date={createdDate} tooltip={{ variant: TimestampTooltipVariant.default }}>
+            {createdDate ? relativeTime(Date.now(), createdDate.getTime()) : 'Unknown'}
+          </Timestamp>
+        </span>
+      </Td>
+      <Td dataLabel="Enabled">
+        <Flex style={{ minWidth: 150 }} gap={{ default: 'gapSm' }}>
+          <span style={{ top: 4 }}>
+            <Switch
+              id={`connection-type-enabled-toggle-${obj.metadata.name}`}
+              isChecked={obj.metadata.annotations?.['opendatahub.io/enabled'] === 'true'}
+              aria-label="toggle enabled"
+              onChange={(_ev, enabled) => onUpdateEnabled(enabled)}
+              data-testid="connection-type-enable-switch"
+            />
```

Comment:
Is this a pattern we use?
![image](https://github.com/user-attachments/assets/01040c96-ad78-4625-9b46-a712bc36771d)

Also I can continuously click on this and send request after request even though the first one has yet to complete.

Reply:
on the Serving Runtimes page, we just disable the switch when the request is pending